esp/esp/users/models/__init__.py

```diff
@@ -922,14 +935,29 @@ def getGrade(self, program = None):
                 regProf = RegistrationProfile.getLastForProgram(self,program)
             if regProf and regProf.student_info:
                 if regProf.student_info.graduation_year:
-                    grade =  ESPUser.gradeFromYOG(regProf.student_info.graduation_year, ESPUser.current_schoolyear(program))
-                    if program:
-                        grade += program.incrementGrade() # adds 1 if appropriate tag is set; else does nothing
+                    return regProf.student_info.graduation_year
+        return None
+    getYOG.get_or_create_token(('self',))
+    getYOG.depend_on_row(lambda: StudentInfo, lambda info: {'self': info.user})
 
+    @cache_function
+    def getGrade(self, program=None, assume_student=False):
+        # assume_student will save us a database hit if the user is a student,
+        # but cost us at least one and possibly several if they're not.  See
+        # ESPUser.getYOG above.
```

Comment:
nitpick: If you change this to a docstring, don't use "above" or "below" when referring to other functions.